includes/class-wcpdf-frontend.php

```diff
@@ -79,6 +80,50 @@ public function woocommerce_api_invoice_number ( $data, $order ) {
 		$this->restore_storing_document_settings();
 	}
 
+	/**
+	 * Download invoice frontend shortcode
+	 */
+	public function download_invoice_shortcode( $atts ) {
+		
+		if( is_admin() ) return;
+
+		// Default values
+		$values = shortcode_atts(array(
+			'order_id'		=> '',
+			'link_text'		=> ''
+		), $atts);
+		if( !$values ) return;
+
+		// Get $order
+		if( is_checkout() && !empty(is_wc_endpoint_url('order-received')) && empty($values['order_id']) ) {
+			$order = wc_get_order( $_GET['order-received'] );
+		} elseif( is_account_page() && !empty(is_wc_endpoint_url('view-order')) && empty($values['order_id']) ) {
+			$order = wc_get_order( $_GET['view-order'] );
+		} elseif( !empty($values['order_id']) ) {
+			$order = wc_get_order( $values['order_id'] );
+		}
+		if( !is_object($order) || !isset($order) || empty($order) ) return;
+
+		// Link text
+		$link_text = __('Download invoice (PDF)', 'woocommerce-pdf-invoices-packing-slips');
+		if( ! empty($values['link_text']) ) {
+			$link_text = $values['link_text'];
+		}
+
+		// User permissions
+		$debug_settings = get_option('wpo_wcpdf_settings_debug');
```

Comment:
@alexmigf I would set default to array so that `isset($debug_settings['guest_access'])` will not throw any notices
e.d.
```php
$debug_settings = get_option('wpo_wcpdf_settings_debug',array());
```